oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCHistoryDeleteDAO.java

```diff
@@ -18,66 +18,100 @@
-                        break;
-                    default:
-                        return;
-                }
+        try (final var conn = jdbcClient.getConnection();
+             final var result = conn.getMetaData().getTables(null, null, TableHelper.getTableName(model) + "%", new String[]{"TABLE"})) {
+            while (result.next()) {
+                tablesToDrop.add(result.getString("TABLE_NAME"));
             }
-            jdbcClient.executeUpdate(connection, dataDeleteSQL.toString(), deadline, minTime);
-            // Delete additional tables
-            for (SQLDatabaseModelExtension.AdditionalTable additionalTable : model.getSqlDBModelExtension()
-                                                                                  .getAdditionalTables()
-                                                                                  .values()) {
-                SQLBuilder additionalTableDeleteSQL = new SQLBuilder("delete from " + additionalTable.getName() + " where ")
-                    .append(timeBucketColumnName).append("<= ? ")
-                    .append(" and ")
-                    .append(timeBucketColumnName).append(">= ? ");
-                jdbcClient.executeUpdate(connection, additionalTableDeleteSQL.toString(), deadline, minTime);
+        }
+
+        ttlTables.forEach(tablesToDrop::remove);
+        tablesToDrop.removeIf(it -> !it.matches(".*_\\d{8}"));
+        for (String table : tablesToDrop) {
+            final var dropSql = new SQLBuilder("drop table if exists ").append(table);
+            jdbcClient.executeUpdate(dropSql.toString());
+        }
+
+        long deadline;
+        long minTime;
+        if (model.isRecord()) {
+            deadline = Long.parseLong(new DateTime().plusDays(-ttl).toString("yyyyMMddHHmmss"));
+            minTime = 1000_00_00_00_00_00L;
+        } else {
+            switch (model.getDownsampling()) {
+                case Minute:
+                    deadline = Long.parseLong(new DateTime().plusDays(-ttl).toString("yyyyMMddHHmm"));
+                    minTime = 1000_00_00_00_00L;
+                    break;
+                case Hour:
+                    deadline = Long.parseLong(new DateTime().plusDays(-ttl).toString("yyyyMMddHH"));
+                    minTime = 1000_00_00_00L;
+                    break;
+                case Day:
+                    deadline = Long.parseLong(new DateTime().plusDays(-ttl).toString("yyyyMMdd"));
+                    minTime = 1000_00_00L;
+                    break;
+                default:
+                    return;
             }
-        } catch (JDBCClientException | SQLException e) {
-            throw new IOException(e.getMessage(), e);
+        }
+        // Delete data in additional tables
+        for (final var additionalTable : model.getSqlDBModelExtension().getAdditionalTables().values()) {
+            SQLBuilder additionalTableDeleteSQL = new SQLBuilder("delete from " + additionalTable.getName() + " where ")
```

Comment:
Why `delete` rather than `drop`? I think the additional table should follow the timeseries process policy.

Reply:
Done